cmd/certsuite/upload/results_spreadsheet/results_spreadsheet.go

```diff
@@ -0,0 +1,367 @@
+			// avoid line breaks in cell
+			val = strings.ReplaceAll(strings.ReplaceAll(col, "\r\n", " "), "\n", " ")
+
+			rowData = append(rowData, &sheets.CellData{
+				UserEnteredValue: &sheets.ExtendedValue{StringValue: &val},
+			})
+		}
+		rows = append(rows, &sheets.RowData{Values: rowData})
+	}
+	return rows
+}
+
+func createSingleWorkloadRawResultsSheet(rawResultsSheet *sheets.Sheet, workloadName string) (*sheets.Sheet, error) {
+	// Initialize sheet with headers
+	filteredRows := []*sheets.RowData{{Values: []*sheets.CellData{
+		{UserEnteredValue: &sheets.ExtendedValue{StringValue: stringToPointer("Owner/TechLead Conclusion")}},
+		{UserEnteredValue: &sheets.ExtendedValue{StringValue: stringToPointer("Next Step Actions")}},
+	}}}
+	filteredRows[0].Values = append(filteredRows[0].Values, rawResultsSheet.Data[0].RowData[0].Values...)
+
+	headers := getHeadersFromSheet(rawResultsSheet)
+	indices, err := getHeaderIndicesByColumnNames(headers, []string{"CNFName"})
+	if err != nil {
+		return nil, err
+	}
+	workloadIndex := indices[0]
+
+	// add to sheet only rows of given workload name
+	for _, row := range rawResultsSheet.Data[0].RowData[1:] {
+		if len(row.Values) > workloadIndex {
+			curWorkloadName := *row.Values[workloadIndex].UserEnteredValue.StringValue
+			if curWorkloadName == workloadName {
+				// add empty values in 2 added columns
+				newRow := &sheets.RowData{
+					Values: append([]*sheets.CellData{{}, {}}, row.Values...),
+				}
+				filteredRows = append(filteredRows, newRow)
+			}
+		}
+	}
+
+	workloadResultsSheet := &sheets.Sheet{
+		Properties: &sheets.SheetProperties{
+			Title: "results",
+		},
+		Data: []*sheets.GridData{{RowData: filteredRows}},
+	}
+
+	return workloadResultsSheet, nil
+}
+
+func createSingleWorkloadRawResultsSpreadSheet(sheetService *sheets.Service, driveService *drive.Service, folder *drive.File, rawResultsSheet *sheets.Sheet, workloadName string) (*sheets.Spreadsheet, error) {
+	workloadResultsSheet, err := createSingleWorkloadRawResultsSheet(rawResultsSheet, workloadName)
+	if err != nil {
+		return nil, err
+	}
+
+	workloadResultsSpreadsheet := &sheets.Spreadsheet{
+		Properties: &sheets.SpreadsheetProperties{
+			Title: fmt.Sprintf("%s Best Practices Test Results", workloadName),
+		},
+		Sheets: []*sheets.Sheet{workloadResultsSheet},
+	}
+
+	workloadResultsSpreadsheet, err = sheetService.Spreadsheets.Create(workloadResultsSpreadsheet).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := addFilterByFailedAndMandatoryToSheet(sheetService, workloadResultsSpreadsheet, "results"); err != nil {
+		return nil, err
+	}
+
+	if err := MoveSpreadSheetToFolder(driveService, folder, workloadResultsSpreadsheet); err != nil {
+		return nil, err
+	}
+
+	return workloadResultsSpreadsheet, nil
+}
+
+//nolint:funlen
+func createConclusionsSheet(sheetsService *sheets.Service, driveService *drive.Service, rawResultsSheet *sheets.Sheet) (*sheets.Sheet, error) {
+	headers := getHeadersFromSheet(rawResultsSheet)
```

Comment:
Suggestion: `headers` --> `rawSheetHeaders`. Then this will help to distinguish it from the `conclusionSheetHeaders`.